modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -242,99 +241,147 @@ public SCIMResponse listWithGET(UserManager userManager, String filter, int star
+            JSONEncoder encoder = getEncoder();
+
+            // Get the URIs of required attributes which must be given a value
+            Map<String, Boolean> requiredAttributes = ResourceManagerUtil
+                    .getOnlyRequiredAttributesURIs((SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes,
+                            excludeAttributes);
+
+            // API group should pass a user manager to GroupResourceEndpoint.
+            if (userManager != null) {
+                List<Object> tempList = userManager
+                        .listGroupsWithGET(rootNode, startIndex, count, sortBy, sortOrder, domainName,
+                                requiredAttributes);
+                return processGroupList(tempList, encoder, attributes, excludeAttributes, startIndex);
+            } else {
+                String error = "Provided user manager handler is null.";
+                // Log the error as well.
+                // Throw internal server error.
+                throw new InternalErrorException(error);
             }
+        } catch (CharonException | NotFoundException | InternalErrorException | BadRequestException |
+                NotImplementedException e) {
+            return encodeSCIMException(e);
+        } catch (IOException e) {
+            String error = "Error in tokenization of the input filter";
+            CharonException charonException = new CharonException(error);
+            return AbstractResourceManager.encodeSCIMException(charonException);
+        }
+    }
 
-            // unless configured returns core-user schema or else returns extended user schema)
-            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
-            if (filter != null) {
-                filterTreeManager = new FilterTreeManager(filter, schema);
-                rootNode = filterTreeManager.buildTree();
+    /**
+     * Resolves the sorting order of the filter.
+     *
+     * @param sortOrder Sort order in the request.
+     * @param sortBy    SortBy in the request.
+     * @return Resolved sorting order.
+     * @throws BadRequestException Invalid sorting order.
+     */
+    private String resolveSortOrder(String sortOrder, String sortBy) throws BadRequestException {
+
+        // Check whether provided sortOrder is valid or not.
+        if (sortOrder != null) {
+            if (!(sortOrder.equalsIgnoreCase(SCIMConstants.OperationalConstants.ASCENDING) || sortOrder
+                    .equalsIgnoreCase(SCIMConstants.OperationalConstants.DESCENDING))) {
+                String error = " Invalid sortOrder value is specified";
+                throw new BadRequestException(error, ResponseCodeConstants.INVALID_VALUE);
             }
+        }
+        // If a value for "sortBy" is provided and no "sortOrder" is specified, "sortOrder" SHALL default to
+        // ascending.
+        if (sortOrder == null && sortBy != null) {
```

Comment:
shall we use StringUtils.isBlank here

Reply:
apache-commons in not in Charon. Therefore, unable to user StringUtils.